src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -206,24 +208,65 @@ public void SuppressOutput()
         }
 
         /// <summary>
-        /// A delegate used to execute children asynchronously.
+        /// Executes children asynchronously and returns their rendered content.
         /// </summary>
         /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
-        /// <remarks>This method is memoized.</remarks>
+        /// <remarks>
+        /// This method is memoized. Multiple calls will not cause children to re-execute with the page's original
+        /// <see cref="HtmlEncoder"/>.
+        /// </remarks>
         public Task<TagHelperContent> GetChildContentAsync()
         {
-            return GetChildContentAsync(useCachedResult: true);
+            return GetChildContentAsync(useCachedResult: true, encoder: null);
         }
 
         /// <summary>
-        /// A delegate used to execute children asynchronously.
+        /// Executes children asynchronously and returns their rendered content.
         /// </summary>
-        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
-        /// of child content; cached content will be returned.</param>
+        /// <param name="useCachedResult">
+        /// If <c>true</c>, multiple calls will not cause children to re-execute with the page's original
+        /// <see cref="HtmlEncoder"/>; returns cached content.
+        /// </param>
         /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
         public Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
         {
-            return _getChildContentAsync(useCachedResult);
+            return GetChildContentAsync(useCachedResult, encoder: null);
+        }
+
+        /// <summary>
+        /// Executes children asynchronously with the given <paramref name="encoder"/> in scope and returns their
+        /// rendered content.
+        /// </summary>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use when the page handles non-<see cref="IHtmlContent"/> C# expressions.
+        /// If <c>null</c>, executes children with the page's current <see cref="HtmlEncoder"/>.
+        /// </param>
+        /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
+        /// <remarks>
+        /// This method is memoized. Multiple calls with the same <see cref="HtmlEncoder"/> instance will not cause
```

Comment:
That is made fairly explicit by `with the same <see cref="HtmlEncoder"/> instance`.